### PR TITLE
Added an easy way to debug objects

### PR DESCRIPTION
**Pull request by kyrena**
### Description

This PR is an old idea to easily debug any `Varien_Object`.

For example in a template: `<?php echo $this ?>` / `<?php echo $product ?>`
When dev mode is enabled, the webpage display:
- `warning_echo_object[Mage_Catalog_Block_Product_View]`
- `warning_echo_object[Mage_Catalog_Model_Product#750]`

Instead of: `Object of class ... could not be converted to string`.
I'm not saying it's very useful.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All automated tests passed successfully (all builds are green)
 - [ ] Add yourself to contributors list

### CODE REVIEW

File: lib/Varien/Object.php [hunk 1]
```diff
@@ -854,4 +854,21 @@ public function flagDirty($field, $flag = true)
...
+     */
+    public function __toString()
+    {
+        if (Mage::getIsDeveloperMode()) {
+            if (PHP_SAPI != 'cli') {
+                return empty($this->getId()) ? '<div><b>warning_echo_object[' . get_class($this) . ']</b></div>' :
+                    '<div><b>warning_echo_object[' . get_class($this) . '#' . $this->getId() . ']</b></div>';
+            }
+            return empty($this->getId()) ? "\n" . 'warning_echo_object[' . get_class($this) . ']' . "\n" :
+                "\n" . 'warning_echo_object[' . get_class($this) . '#' . $this->getId() . ']' . "\n";
```
Comment (fballiano) [inline lib/Varien/Object.php:870]:
```suggestion
            if (PHP_SAPI == 'cli') {
                return empty($this->getId()) ? "\n" . 'warning_echo_object[' . get_class($this) . ']' . "\n" :
                    "\n" . 'warning_echo_object[' . get_class($this) . '#' . $this->getId() . ']' . "\n";
            }
            return empty($this->getId()) ? '<div><b>warning_echo_object[' . get_class($this) . ']</b></div>' :
                '<div><b>warning_echo_object[' . get_class($this) . '#' . $this->getId() . ']</b></div>';
```

I think == ifs are more readable than != one :-)

File: lib/Varien/Object.php [hunk 2]
```diff
@@ -854,4 +854,21 @@ public function flagDirty($field, $flag = true)
         }
         return $this;
     }
+
+    /**
+     * @throw Exception
+     * return string
+     */
+    public function __toString()
```
Comment (fballiano) [inline lib/Varien/Object.php:862]: I like the idea a lot, I was wondering, is it possible that somebody is already casting a varien_object to string for some reason (or mistake) and we're braking something?

Comment (luigifab) [reply lib/Varien/Object.php:862]: I suppose that if you cast to string any object that extends `Varien_Object`, you have your custom `__toString()` method? But I saw that PHPStan is unhappy.

Comment (fballiano) [reply lib/Varien/Object.php:862]: very weird, doesn't seem to be related to this PR...

**Comment by pquerner** [comment]
I dont see the point in this really.
However, what I can see it adding something like this https://github.com/symfony/var-dumper if you *really* need to debug like that.

**Comment by fballiano** [comment]
ye, I also think that probably the symfony composer is a better idea, and it's for sure more complete and feature rich.

for this reason I'll close this PR, so sorry! 😞

**Pull request closed without merging**